teps/0040-ignore-step-errors.md

```diff
@@ -145,13 +156,304 @@ This proposal is limited to a step within a task and does not address `pipelineT
+container termination state.
+
+To ignore a step error, set the `onError` to `continue`:
+
+```yaml
+steps:
+  - image: docker.io/library/golang:latest
+    name: ignore-unit-test-failure
+    exit:
+      onError: continue
+    script: |
+      go test .
+```
+
+A `step` with `onError` set to `continue` does not fail the `taskRun` and continues executing the rest of the steps in
+a task. The original failed exit code of the wrapped command is available in the termination state of the container.
+
+```
+kubectl get tr taskrun-unit-test-t6qcl -o json | jq .status
+{
+  "completionTime": "2021-06-21T18:22:06Z",
+  "conditions": [
+    {
+      "lastTransitionTime": "2021-06-21T18:22:06Z",
+      "message": "All Steps have completed executing",
+      "reason": "Succeeded",
+      "status": "True",
+      "type": "Succeeded"
+    }
+  ],
+  "podName": "taskrun-unit-test-t6qcl-pod-zpqs9",
+  "startTime": "2021-06-21T18:21:57Z",
+  "steps": [
+    {
+      "container": "step-ignore-unit-test-failure",
+      "imageID": "...",
+      "name": "ignore-unit-test-failure",
+      "terminated": {
+        "containerID": "...",
+        "exitCode": 1,
+        "finishedAt": "2021-06-21T18:22:05Z",
+        "reason": "Completed",
+        "startedAt": "2021-06-21T18:22:05Z"
+      }
+    },
+  ],
+```
+
+As part of this design, we are introducing an additional internal volume `/tekton/steps/`. A file named `exitCode` will
+be created for each step to store the non-zero exit code, for example:
+
+```
+/tekton/steps/<step-name>/exitCode
+```
+
+`<step-name>` will be replaced with the name of the step. If a step does not have any name, `<step-name>` will be
+replaced with `step-unnamed-<step-index>` where `<step-index>` is `0` for the first step, `1` for the second step, so
+on and so forth.
+
+This new internal volume `/tekton/steps/<step-name>/` can be utilized in future to collect any metadata for each step.
+And instead of referencing to it with the `<step-name>`, will create a symlink such that it can be accessed using the
+`<step-index>`.
+
+```shell
+ln -s /tekton/steps/<step-name> /tekton/steps/<step-index>
+```
+
+Any subsequent step can access the non-zero exit code of a previous step using the `path` similar to a task result,
+for example:
+
+```shell
+$(steps.<step-name>.exitCode.path)
+```
+
+The `exitCode` of a step without any name can be referenced using:
+
+```shell
+$(steps.step-unnamed-<step-index>.exitCode.path)
+```
+
+If you would like to use the tekton internal path, you can access the exit code by reading the file
+(it is not recommended though):
+
+```shell
+cat /tekton/steps/<step-name>/exitCode
+```
+
+In the following example, the `pipelineRun` is executing two tasks with the first task defined to ignore the step error.
+If that step is able to produce a result file before failing, that result is made available to it's consuming task.
+
+#### the task result with a failing step
+
+```
```

Comment:
might be worth adding a description to explain this example - im assuming this is showing that you can have a failed step, but still produce a result?

Reply:
yup its showing that its possible to have a failed step but still produce a result, added some explanation.